### PR TITLE
Add generics to Java -> ObjC translator.

### PR DESCRIPTION
**Pull request by corvino**

### CODE REVIEW

File: tools/objc/base.py [hunk 1]
```diff
@@ -100,9 +101,21 @@ def convert_type(config, package, type):
                     "native": False
                 }
             else:
+                generic_spec = ""
+                if objc_type_name in config.data["generics"]:
+                    for generic_type in type_info.type_arguments:
+                        spec_type = BaseTranslator.convert_type(config, package, generic_type)
+                        generic_spec = generic_spec + ", " + spec_type["objc_type"]
+
+                if 2 < len(generic_spec):
```
Comment (rasekhg) [inline tools/objc/base.py]: Better to create a list and then ','.join(string_list) to put in the commas.

Comment (corvino) [inline tools/objc/base.py]: Yes, much cleaner. Thanks.

File: tools/objc/base.py [hunk 2]
```diff
@@ -74,6 +74,7 @@ def convert_types(config, package, type):
 
     @staticmethod
     def convert_type(config, package, type):
+        type_info = type
```
Comment (egorpushkin) [inline tools/objc/base.py:77]: What is the purpose of this declaration?

Comment (corvino) [inline tools/objc/base.py:77]: It's to work around the potential reuse of type when 'type = type.name.value' is executed. I still need access type.type_arguments later on.

Rasekh had the same question, and I didn't intend this to necessarily be the optimal refactor. But there a number of ways to go, so this at least raises the question. Let me know what you want to do.

Comment (egorpushkin) [inline tools/objc/base.py:77]: I see now. Missed that

**Comment by egorpushkin** [comment]
LGTM

**Pull request merged**